src/cc/libbpf.c

```diff
@@ -1036,51 +929,128 @@ static void exit_mount_ns(int fd) {
-                      uint64_t offset, pid_t pid)
+static int create_probe_event(char *buf, const char *ev_name,
+                              enum bpf_probe_attach_type attach_type,
+                              const char *config1, uint64_t offset,
+                              const char *event_type, pid_t pid, int maxactive)
 {
-  char buf[PATH_MAX];
-  char event_alias[PATH_MAX];
-  static char *event_type = "uprobe";
-  int res, kfd = -1, pfd = -1, ns_fd = -1;
-  // Try create the uprobe Perf Event with perf_event_open API.
-  pfd = bpf_try_perf_event_open_with_probe(binary_path, offset, pid, event_type,
-                                           attach_type != BPF_PROBE_ENTRY);
-  // If failed, most likely Kernel doesn't support the new perf_event_open API
-  // yet. Try create the event using debugfs.
-  if (pfd < 0) {
-    snprintf(buf, sizeof(buf), "/sys/kernel/debug/tracing/%s_events", event_type);
-    kfd = open(buf, O_WRONLY | O_APPEND, 0);
-    if (kfd < 0) {
-      fprintf(stderr, "open(%s): %s\n", buf, strerror(errno));
-      goto error;
-    }
+  int kfd = -1, res = -1, ns_fd = -1;
+  char ev_alias[128];
+  bool is_kprobe = strncmp("kprobe", event_type, 6) == 0;
 
-    res = snprintf(event_alias, sizeof(event_alias), "%s_bcc_%d", ev_name, getpid());
-    if (res < 0 || res >= sizeof(event_alias)) {
-      fprintf(stderr, "Event name (%s) is too long for buffer\n", ev_name);
-      goto error;
-    }
-    res = snprintf(buf, sizeof(buf), "%c:%ss/%s %s:0x%lx", attach_type==BPF_PROBE_ENTRY ? 'p' : 'r',
-                   event_type, event_alias, binary_path, (unsigned long)offset);
-    if (res < 0 || res >= sizeof(buf)) {
-      fprintf(stderr, "Event alias (%s) too long for buffer\n", event_alias);
-      goto error;
-    }
+  snprintf(buf, PATH_MAX, "/sys/kernel/debug/tracing/%s_events", event_type);
+  kfd = open(buf, O_WRONLY | O_APPEND, 0);
+  if (kfd < 0) {
+    fprintf(stderr, "%s: open(%s): %s\n", __func__, buf,
+            strerror(errno));
+    return -1;
+  }
 
-    ns_fd = enter_mount_ns(pid);
-    if (write(kfd, buf, strlen(buf)) < 0) {
-      if (errno == EINVAL)
-        fprintf(stderr, "check dmesg output for possible cause\n");
-      goto error;
+  res = snprintf(ev_alias, sizeof(ev_alias), "%s_bcc_%d", ev_name, getpid());
+  if (res < 0 || res >= sizeof(ev_alias)) {
+    fprintf(stderr, "Event name (%s) is too long for buffer\n", ev_name);
+    close(kfd);
+    goto error;
+  }
+
+  if (is_kprobe) {
+    if (offset > 0 && attach_type == BPF_PROBE_ENTRY)
+      snprintf(buf, PATH_MAX, "p:kprobes/%s %s+%"PRIu64,
+               ev_alias, config1, offset);
+    else if (maxactive > 0 && attach_type == BPF_PROBE_RETURN)
+      snprintf(buf, PATH_MAX, "r%d:kprobes/%s %s",
+               maxactive, ev_alias, config1);
+    else
+      snprintf(buf, PATH_MAX, "%c:kprobes/%s %s",
+               attach_type == BPF_PROBE_ENTRY ? 'p' : 'r',
+               ev_alias, config1);
+  }
+  else {
```

Comment:
Please fix style to be consistent within the file:)

Reply:
@palmtenor Thanks for the review!
Since I found nothing about coding style like `astyle`, `.clang-format` or any docs,  I only simulate the style in the file.